src/cmd/sv-rollout/svrestarter.go

```diff
@@ -58,26 +61,28 @@ func (s *SvRestarter) Restart() error {
 	case <-restartDone:
 		if err != nil {
 			if strings.Contains(string(out), "timeout: run: ") {
-				err = ErrRestartTimeout{Service: s.Service}
+				rerr = ErrRestartTimeout{Service: s.Service}
 				tags = append(tags, "status:timeout")
 			} else {
-				err = ErrRestartFailed{Service: s.Service, Message: string(out)}
+				rerr = ErrRestartFailed{Service: s.Service, Message: string(out)}
 				tags = append(tags, "status:success")
 			}
 		}
 	case <-s.preempt:
 		<-preemptionAcceptable
-		err = ErrRestartPreempted{Service: s.Service}
+		rerr = ErrRestartPreempted{Service: s.Service}
 		tags = append(tags, "status:preempted")
 	}
 
-	if Statsd != nil {
-		tags = append(tags, "service:"+s.Service)
-		Statsd.Timer("service.restart", time.Since(start), tags, 1)
-	}
+	go func() {
```

Comment:
I think creating a goroutine here is more expensive than the inline code, why os the goroutine better ?